src/file_pipe_log/log_file.rs

```diff
@@ -6,35 +6,114 @@ use std::io::{Read, Seek, SeekFrom, Write};
 use std::sync::Arc;
 
 use crate::metrics::*;
-use crate::pipe_log::FileBlockHandle;
+use crate::pipe_log::{FileBlockHandle, FileSeq};
 use crate::{Error, Result};
 
-use super::format::LogFileHeader;
+use super::format::{LogFileHeader, Version};
 use crate::env::{FileSystem, Handle, WriteExt};
 
 /// Maximum number of bytes to allocate ahead.
 const FILE_ALLOCATE_SIZE: usize = 2 * 1024 * 1024;
 
+/// Combination of `[Handle]` and `[Version]`, specifying a handler of a file.
+#[derive(Clone, Debug)]
+pub struct FileHandler<F: FileSystem> {
+    pub seq: FileSeq,
+    pub handle: Arc<F::Handle>,
+    pub version: Option<Version>,
+}
+
+/// Function for reading the header of the log file, and return a
+/// `[LogFileHeader]`.
+///
+/// Attention please, to avoid to move the offset of the given `[handle]`, we
+/// use a copy of the `[handle]` to parse the header of the given file.
+pub(super) fn build_file_header<F: FileSystem>(
+    system: &F,
+    handle: Arc<F::Handle>,
+    expected_reader: Option<&mut F::Reader>,
+) -> Result<LogFileHeader> {
```

Comment:
I think too many things are going on in this function. You can make this function a member of `LogFileReader`. That seems more natural to me.

Reply:
You don't need to deal with file system when implementing member function for `LogFileReader`. It is a utility wrapper over the reader produced by file system.

Reply:
Got it, i did know what u meant.

Reply:
This comment is not addressed. I expect it to be changed to `LogFileReader::parse_header(&mut self)`.